app/forms/Transport.scala

```diff
@@ -42,20 +42,45 @@ object Transport {
-    Seq(Sea, Rail, Road, Air, PostalOrMail, FixedInstallations, InlandWaterway, Other)
-
-  val mapping = Forms.mapping(
-    "modeOfTransport" -> requiredRadio("transport.modeOfTransport.empty")
-      .verifying("transport.modeOfTransport.error", isContainedIn(allowedModeOfTransport)),
-    "nationality" -> text()
-      .verifying("transport.nationality.empty", nonEmpty)
-      .verifying("transport.nationality.error", isEmpty or isValidCountryCode),
-    "transportId" -> text()
-      .verifying("transport.transportId.empty", nonEmpty)
-      .verifying("transport.transportId.error", isEmpty or (noLongerThan(35) and isAlphanumericWithAllowedSpecialCharacters))
-  )(Transport.apply)(Transport.unapply)
-
-  def form: Form[Transport] =
-    Form(mapping)
+  private def form2Model(modeOfTransport: String, nationality: String, transportId: String): Transport =
+    Transport(modeOfTransport = Some(modeOfTransport), nationality = Some(nationality), transportId = Some(transportId))
+
+  private def model2Form(transport: Transport): Option[(String, String, String)] =
+    for {
+      modeOfTransport <- transport.modeOfTransport
+      nationality <- transport.nationality
+      transportId <- transport.transportId
+    } yield (modeOfTransport, nationality, transportId)
+
+  val allowedModesOfTransport: Set[String] =
+    Set(Sea, Rail, Road, Air, PostalOrMail, FixedInstallations, InlandWaterway, Other)
+
+  private val outOfTheUkMapping = Forms
+    .mapping(
+      "modeOfTransport" -> requiredRadio("transport.modeOfTransport.empty")
+        .verifying("transport.modeOfTransport.error", isContainedIn(allowedModesOfTransport)),
+      "nationality" -> text()
+        .verifying("transport.nationality.empty", nonEmpty)
+        .verifying("transport.nationality.error", isEmpty or isValidCountryCode),
+      "transportId" -> text()
+        .verifying("transport.transportId.empty", nonEmpty)
+        .verifying("transport.transportId.error", isEmpty or (noLongerThan(35) and isAlphanumericWithAllowedSpecialCharacters))
+    )(form2Model)(model2Form)
+
+  private val atLeastOneIsEmpty: Transport => Boolean = (t: Transport) => t.transportId.isEmpty || t.nationality.isEmpty || t.modeOfTransport.isEmpty
+
+  private val backIntoTheUkMapping = Forms
+    .mapping(
+      "modeOfTransport" -> optional(text().verifying("transport.modeOfTransport.error", isContainedIn(allowedModesOfTransport))),
+      "nationality" -> optional(text().verifying("transport.nationality.error", isEmpty or isValidCountryCode)),
+      "transportId" -> optional(
+        text().verifying("transport.transportId.error", isEmpty or (noLongerThan(35) and isAlphanumericWithAllowedSpecialCharacters))
+      )
+    )(Transport.apply)(Transport.unapply)
+    .verifying("transport.backIntoTheUk.error.allFieldsEntered", atLeastOneIsEmpty)
+
+  def form: Form[Transport] = Form(outOfTheUkMapping)
+
+  def outOfTheUkForm: Form[Transport] = Form(outOfTheUkMapping)
```

Comment:
Do you need both `form` and `outOfTheUkForm`?

Reply:
Nope. Good catch! 👍